android_app/app/src/main/java/com/health/openscale/core/export/OpenScaleContentProvider.java

```diff
@@ -0,0 +1,129 @@
+/* Copyright (C) 2018 Paul Cowan <paul@custardsource.com>
+ *
+ *    This program is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>
+ */package com.health.openscale.core.export;
+
+import android.content.ContentProvider;
+import android.content.ContentUris;
+import android.content.ContentValues;
+import android.content.Context;
+import android.content.UriMatcher;
+import android.database.Cursor;
+import android.net.Uri;
+import android.text.TextUtils;
+import android.util.Log;
+
+import com.health.openscale.core.OpenScale;
+
+/**
+ * Exposes the user and measurement data from openScale via
+ * <a href="https://developer.android.com/guide/topics/providers/content-providers">Android
+ * Content Providers</a>. This allows other apps to access the openScale data for their own purposes
+ * (e.g. syncing to third-party services like Google Fit, Fitbit API, etc) without openScale itself
+ * needing to do so or request additional permissions. <br />
+ *
+ * This access is gated by the com.health.openscale.READ_DATA permission, which is defined in the
+ * manifest; it is not accessible to any other app without user confirmation.<br />
+ *
+ * The following URIs are supported:
+ * <ul>
+ *     <li><code>content://com.health.openscale.provider/user</code>: list all users.</li>
+ *     <li><code>content://com.health.openscale.provider/user/$ID</code>: retrieve single user
+ *         by ID.</li>
+ *     <li><code>content://com.health.openscale.provider/user/$ID/measurements</code>:
+ *         retrieve all measurements for the supplied user ID.</li>
+ * </ul>
+ */
+public class OpenScaleContentProvider extends ContentProvider {
+    private static final UriMatcher uriMatcher = new UriMatcher(UriMatcher.NO_MATCH);
+
+    private static final String AUTHORITY = "com.health.openscale.provider";
```

Comment:
Use BuildConfig.APPLICATION_ID instead of hard coding "com.health.openscale". Here and in all other places.